src/components/static-map.js

```diff
@@ -31,6 +32,7 @@ import mapboxgl from '../utils/mapboxgl';
 import {checkVisibilityConstraints} from '../utils/map-constraints';
 import {MAPBOX_LIMITS} from '../utils/map-state';
 import MapContext from './map-context';
+import {diff as diffStyles} from '@mapbox/mapbox-gl-style-spec';
```

Comment:
This package is published from https://github.com/mapbox/mapbox-gl-js/tree/master/src/style-spec and is exactly what mapbox uses to do diffing:
https://github.com/mapbox/mapbox-gl-js/blob/master/src/style/style.js#L30
https://github.com/mapbox/mapbox-gl-js/blob/master/src/style/style.js#L435

If this issue is caused by a bug in mapbox-gl it should be fixed in their repo.

Reply:
I believe this is a difference in behavior between the two libraries. mapbox-gl chooses to error if it cannot update the style just yet. In our case, we want to update the map for the user when we can, retrying if needed.

Reply:
If the error is just caused by map not being loaded, why do we need to implement our own diffing logic? Can we just call `map.setStyle` when it's ready?

Reply:
hrm. good point

Reply:
I've looked at [this part of code](https://github.com/mapbox/mapbox-gl-js/blob/master/src/style/style.js#L435) with debugger and found that 
this.serialize() is very different to what is nextState, using simple diff tool.

It generates thousands of changes for exactly same style layers every time.
Do we need to normalize nextState somehow before diffing?

In this code change, we are diffing styles without mutations, so it only detects real changes